### PR TITLE
TRUNK-6231: Migrate from Hibernate Mapping XML to JPA annotations.

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

Migrated the following entities from Hibernate mapping XML to JPA annotations:

- DrugIngredient
- FieldAnswer
- AlertRecipient

## Issue I worked on

see https://openmrs.atlassian.net/browse/TRUNK-6231

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/AlertRecipientPK.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import javax.persistence.Embeddable;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Embeddable
+public class AlertRecipientPK implements Serializable {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/AlertRecipientPK.java:17]: What is this new class for?

**Comment by dkayiwa** [comment]
@wikumChamith you have not yet responded to my comment.

**Comment by wikumChamith** [comment]
> @wikumChamith you have not yet responded to my comment.

Screenshot from 2024-05-19 19-06-14

Am I missing something :sweat_smile:

**Comment by dkayiwa** [comment]
Your screenshot has `pending`

**Comment by wikumChamith** [comment]
I replied to your question before. In case you are not seeing it, I have my answer here.

> To replace this composite key.
> 
> ```
> <composite-id>
> 			<key-many-to-one name="alert" column="alert_id"	class="org.openmrs.notification.Alert" />
> 			<key-many-to-one name="recipient" class="org.openmrs.User" column="user_id" />
> </composite-id>
> ```

**Comment by dkayiwa** [comment]
Can't we pull this off without creating the new class?

**Comment by wikumChamith** [comment]
I initially followed the recommended approach by creating a separate class. However, we can achieve this without creating an additional class. I have updated the PR accordingly.

https://www.baeldung.com/jpa-composite-primary-keys

**Comment by dkayiwa** [comment]
> I initially followed the [recommended approach](https://www.baeldung.com/jpa-composite-primary-keys) by creating a separate class. However, we can achieve this without creating an additional class. I have updated the PR accordingly.

Did you test these changes by running the webapp and  confirm that all is well?

**Comment by dkayiwa** [comment]
@wikumChamith i also find the title not clear enough. What are you migrating? Is it all the classes that are currently mapped to hibernate mapping xml files? In the past, we have done this as one entity at a time. That is, each with its own ticket and pull request.

**Comment by wikumChamith** [comment]
I agree. I'll try to create sperate tickets and pull requests for each entity.

**Comment by wikumChamith** [comment]
I am closing this PR to create separate pull requests for each entity.

**Pull request closed without merging**